RFC/src/RFC-0300_DAN.md

```diff
@@ -1 +1,205 @@
+output for a minimum amount of time ([RegistrationTerm]) as well as some metadata, such as the VNs public key and a generated Node ID. The Node ID is generated
+during registration to prevent mining of VN public keys that can be used to manipulate routing on the DAN. The blinding factor for the Registration transaction is the private key
+that the VN node will use to sign every instruction that it executes for the duration of its [RegistrationTerm].
+
+Once a VNs [RegistrationTerm] has expired so will this specific VN registration. The UTXO timelock will have elapsed so the [RegistrationCollateral] can be reclaimed and a new VN registration
+need to be performed. This automatic registration expiry will ensure that the VN registry stays up to date with active VN registrations and inactive registrations will naturally be removed.
+
+Requiring nodes to register themselves serves two purposes:
+* Makes VN Sybil attacks expensive,
+* Provides an authoritative "central-but-not-centralised" registry of validator nodes from the base layer.
+
+#### Execution of instructions
+VNs are expected to manage the state of digital assets on behalf of digital asset issuers. They receive fees as reward
+for doing this.
+* Digital assets consist of an initial state plus a set of state transition rules. These rules are set by the Tari
+  protocol, but will usually provide parameters that must be specified by the [Asset Issuer].
+* The set of VNs that participate in managing state of a specific DA is called a [Committee]. A committee is selected during the asset
+issuance process and membership of the committee can be updated at [Checkpoint]s.
+* It is the VNs responsibility to ensure that every state change in a digital asset conforms to the contract's rules.
+* VNs accept digital asset [Instructions] from clients and peers. [Instructions] allow for creating, updating, and
+  expiring digital assets on the DAN.
+* VNs provide additional collateral when accepting an offer to manage an asset, which is stored in a multi-signature
+  contract on the base layer. This collateral can be taken from the VN if it is proven that the VN engaged in
+  malicious behaviour.
+* VNs participate in fraud proof validations in the event of consensus disputes (which could result in the malicious VN's
+  collateral being slashed).
+* Digital asset metadata (e.g. large images) are managed by VNs, but whether the data is considered part of the state
+  (and thus checkpointed) or out of state depends on the type of digital asset contract employed.
+
+#### Fees
+Fees will be paid to VNs based on the amount of work they did during a checkpoint period. The fees will be paid from a fee pool which will be collected
+and reside in a UTXO that is accessible by the committee. The exact mechanism for the the payment of the fees by the committee and who pays the various
+types of fees is still under discussion.
+
+#### Checkpoints
```

Comment:
I see there is not cut-through or pruning mentioned in the checkpoints. We will need to implement this. 
But we need to answer questions as how long/deep to we save them?

Reply:
I believe that level of detail will be covered in the sub-RFC that deals with the details of the checkpoints, of which there is still much to discuss and decide.